### PR TITLE
VACMS-7557: Enable allow only one on health services.

### LINKED ISSUE #7557: Add governance measures for unique ID for Service taxonomy terms
## User Story or Problem Statement

As a lighthouse engineer for the facility API I need a unique identifier for a service that is more meaningful but as immutable as a term id so that consumers of the API can use semi friendly request names for services instead of term ids.

## Acceptance Criteria
- [ ] ~A new field on health_care_service_taxonomy terms 'field_health_service_uid' (plain text) label: Facility health service UID'~
- [ ]  ~hook_update_N created (must be in separate PR that goes out after the field creation) that populates the field with camel case version of current term name.  (must be stripped of all non-alpha-num characters [no ' " ( ) + - .]~
- [x] #7671
- [x] Disable field_health_service_api_id for all users but admin on health_care_service_taxonomy  (use UserPerms service)
- [x] allow_only_one field is added to  health_care_service_taxonomy so that the `field_health_service_api_id` can never be used twice.
   - [x] Set form display of the allow_only_one field to appear just above  the  field_health_service_api_id field.
- [x] `field_health_service_api_id` is required.
- [ ] Change section for all service taxonomy terms to an OCTO > VA Service taxonomy

## Implementation steps
<optional>

## Design principles

Veteran-centered
- [x] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [ ] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [ ] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [ ] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] `Empowering`: Provide clear information to help editors make decisions about their work.

### CMS Team

Please leave only the team that will do this work selected.
- [ ] `Platform CMS Team`
- [x] `Sitewide CMS Team`
  - [x] `⭐️ Content ops`
  - [ ] `⭐️ CMS experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product support`
  - [ ] `⭐️ User support`

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
closes #7557

## Testing done
Visual/Behat

## Screenshots
image

## QA steps
 - [x] As an administrator, go to `/taxonomy/term/112/edit` and change `Health Service API ID` field to `addiction` and visually verify `1 error has been found: Enforce unique combo service` message appears on save.
 - [x] As a non administrator who has access to health services taxonomy, go to `/taxonomy/term/112/edit` and visually verify  `Health Service API ID` field is disabled.

### CODE REVIEW

File: config/sync/field.field.taxonomy_term.health_care_service_taxonomy.field_health_service_api_id.yml [hunk 1]
```diff
@@ -5,13 +5,19 @@ dependencies:
   config:
     - field.storage.taxonomy_term.field_health_service_api_id
     - taxonomy.vocabulary.health_care_service_taxonomy
+  module:
```
Comment (swirtSJW) [inline config/sync/field.field.taxonomy_term.health_care_service_taxonomy.field_health_service_api_id.yml:8]: This is a residual change that bubbles up when config is saved.  Not expected in this PR, but acceptable.

File: docroot/modules/custom/va_gov_backend/src/EventSubscriber/EntityEventSubscriber.php [hunk 2]
```diff
@@ -293,6 +294,25 @@ public function lockTitleEditing(array &$form, FormStateInterface $form_state) {
...
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public function lockApiIdEditing(array &$form, FormStateInterface $form_state) {
+    $form_object = $form_state->getFormObject();
+    $bundle = NULL;
+    if ($form_object instanceof ContentEntityForm) {
+      $bundle = $form_object->getEntity()->bundle();
+    }
+    if (!$this->userPermsService->hasAdminRole() && $bundle === 'health_care_service_taxonomy') {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/src/EventSubscriber/EntityEventSubscriber.php:311]: This should only be for true admins.
```suggestion
    if (!$this->userPermsService->hasAdminRole(TRUE) && $bundle === 'health_care_service_taxonomy') {
```

**Review by swirtSJW — COMMENTED** [review]
Looks good over all.   NIce work.   Recommending only one change and will commit that directly.

**Review by swirtSJW — APPROVED** [review]
Tests out good.
image

**Comment by swirtSJW** [comment]
image

File: docroot/modules/custom/va_gov_backend/src/EventSubscriber/EntityEventSubscriber.php (hunk 2 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/src/EventSubscriber/EntityEventSubscriber.php]: Ty for catching that, @swirtSJW :heart:

**Pull request merged**